addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -79,11 +81,19 @@ options.registry.TableOfContent = options.Class.extend({
      */
     _generateNav: function (ev) {
         this.options.wysiwyg && this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
+        const navEl = this.$target[0].querySelector('.s_table_of_content_navbar');
         const headingsEls = this.$target.find(this.targetedElements).toArray()
             .filter(el => !el.closest('.o_snippet_desktop_invisible'));
         const areHeadingsEqual = this.oldHeadingsEls.length === headingsEls.length
             && this.oldHeadingsEls.every((el, i) => el.isEqualNode(headingsEls[i]));
-        if (areHeadingsEqual) {
+        const areVisibilityIdsMatching = headingsEls.every((headingEl) => {
+            const visibilityId = headingEl.closest('section').getAttribute('data-visibility-id');
+            const matchingLinkEl = navEl.querySelector(`a[href="#${headingEl.getAttribute('id')}"]`);
+            const matchingLinkVisibilityId = matchingLinkEl ? matchingLinkEl.getAttribute('data-visibility-id') : null;
+            // Check if visibilityId matches matchingLinkVisibilityId or both are null/undefined
+            return visibilityId == matchingLinkVisibilityId;
```

Comment:
```suggestion
            return visibilityId === matchingLinkVisibilityId;
```
![image](https://github.com/odoo/odoo/assets/30048408/2093850f-4458-4342-8888-9c1ce1d2d7f2)

Reply:
I encountered a situation when using JQuery where there was a comparison between null and undefined. The `visibilityId` received a `null` parameter, while `matchingLinkVisibilityId` received `undefined`. When I compared these values using `===`, I didn't get the expected result. To resolve this, I used `==` instead of `===` to compare only the values and not the object types. I forgot that I had switched to JavaScript code. We can use `===` instead of `==` as now we only get `null`.